rotkehlchen/chain/arbitrum_one/modules/hyperliquid/decoder.py

```diff
@@ -0,0 +1,107 @@
+
+import logging
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.arbitrum_one.modules.hyperliquid.constants import (
+    BRIDGE_ADDRESS,
+    CPT_HYPER,
+    FINALIZE_WITHDRAWAL,
+)
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import DEFAULT_DECODING_OUTPUT
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.utils.misc import bytes_to_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.structures import DecoderContext, DecodingOutput
+    from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.types import ChecksumEvmAddress, EvmTransaction
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class HyperliquidDecoder(DecoderInterface):
+    """Hyperliquid deposit/withdrawals work only with USDC"""
+
+    def _process_deposit(
+            self,
+            transaction: 'EvmTransaction',
+            decoded_events: list['EvmEvent'],
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> list['EvmEvent']:
+        """Deposits to Hyperliquid are just transfers to the bridge contract.
+        There aren't any contract call
+        """
+        for event in decoded_events:
+            if (
+                event.address == BRIDGE_ADDRESS and
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == Asset('eip155:42161/erc20:0xaf88d065e77c8cC2239327C5EDb3A432268e5831')  # USDC  # noqa: E501
```

Comment:
Don't have a constructor call each for loop iteration

```suggestion
                event.asset.identifier =='eip155:42161/erc20:0xaf88d065e77c8cC2239327C5EDb3A432268e5831'  # USDC  # noqa: E501
```